### PR TITLE
Add wamr to esp-idf components registry.

### PR DESCRIPTION
**Pull request by donghengqaz**
Espressif has [ESP-IDF component registry](https://components.espressif.com/) to manage external components, and developers can use this directly and easily.

I plan to add `wasm-micro-runtime` to this registry, but it seems that wasm-micro-runtime team will not plan to cost much resource to maintain ESP-IDF platform and its related things, so I discuss with some internal Espressif developers. We hope you approve to let us publish WAMR into this registry with specific version(not WAMR official version). If some patches need to add for ESP-IDF platform, I will raised PR here, and when this PR is merged, I publish new version to registry.

If you want to register account in this registry, please add me into your namespace with necessary access permissions, otherwise we plan to use our own account to publish WAMR.

### CODE REVIEW

File: idf_component.yml
Comment (wenyongh) [inline idf_component.yml:1]: Must this file be put under the root directory? There are too many files under root dir now.

File: CMakeLists.txt
Comment (wenyongh) [inline CMakeLists.txt:6]: How about:
```cmake
if (ESP_PLATFORM)
  include (${COMPONENT_DIR}/build-scripts/esp-idf/wamr/CMakeLists.txt)
  return ()
endif ()
```

File: idf_component.yml
Comment (donghengqaz) [reply idf_component.yml:1]: Yes, this is need to be under the root directory, but version can be WAMR version plus bugfix version, for example now WAMR version is 1.3.2, we can make this to be 1.3.2.x(x is bugfix version is needed).

File: CMakeLists.txt
Comment (donghengqaz) [reply CMakeLists.txt:6]: OK.

**Comment by donghengqaz** [comment]
@wenyongh Today we have a meeting about this, we would like to use release branch of WAMR, because we may use this in real product, so master branch is not good choice, so I plan to merge this into release/v1.3.x version.

**Comment by lum1n0us** [comment]
@donghengqaz here are some questions to make sure we are on the same page.

- Please be awarded of that 1.3.x branch will be out of maintenance in the end of this year(2024).
- I noticed this PR will be merged only in 1.3.x but not in both (main and 1.3.x). If we all hope a smooth upgrade from 1.3.x to 2.x in next year(2025) after termination of 1.3.x branch, it should be better to merge both and maintain both
- Since we are not sure you are the only customer depends on 1.3.x branch, there might be following PRs from them too. I guess a 1.3.x release is better than "the latest commit of 1.3.x branch". So, what's your opinion about make a new 1.3.x release recently？

**Comment by donghengqaz** [comment]
Hi @lum1n0us, 

> I noticed this PR will be merged only in 1.3.x but not in both (main and 1.3.x). If we all hope a smooth upgrade from 1.3.x to 2.x in next year(2025) after termination of 1.3.x branch, it should be better to merge both and maintain both

You are right, and I will add another PR for main. At first we plan to publish version based on WAMR release branch, because some internal developers and customers need a stable version to start developing product software.

> Since we are not sure you are the only customer depends on 1.3.x branch, there might be following PRs from them too. I guess a 1.3.x release is better than "the latest commit of 1.3.x branch". So, what's your opinion about make a new 1.3.x release recently？

Yes, this is very well. I will continue to publish new version to registry, you can see WAMR version in registry as follows:
```
1.3.2-1
```
The `1.3.2` is WAMR tag version based on release version, and `-1` is bugfix version which is new patch 
merged into release branch. If recently WAMR v1.3.3 is released, I will publish it into registry with version `1.3.3-0`.

**Comment by wenyongh** [comment]
@donghengqaz the CI runs failed since the wamr-ide CI issue isn't fixed yet (which is fixed in main branch), I submitted PR https://github.com/bytecodealliance/wasm-micro-runtime/pull/3268 to fix the issues that were fixed in main branch, and will merge your PR after that PR is merged.

**Comment by wenyongh** [comment]
> @donghengqaz I have merged PR #3268, could you rebase it? BTW, could you submit another PR to main branch?

**Review by wenyongh — APPROVED** [review]

**Comment by donghengqaz** [comment]
> > @donghengqaz I have merged PR #3268, could you rebase it? BTW, could you submit another PR to main branch?

OK. In the main branch, which version in file `idf_component.yml` can I use, is `1.3.3` OK?

**Comment by wenyongh** [comment]
> > > @donghengqaz I have merged PR #3268, could you rebase it? BTW, could you submit another PR to main branch?
> 
> OK. In the main branch, which version in file `idf_component.yml` can I use, is `1.3.3` OK?

The next planned release version in main is 2.0.0, maybe you can use it in idf_component.yml?

**Comment by donghengqaz** [comment]
> > > > @donghengqaz I have merged PR #3268, could you rebase it? BTW, could you submit another PR to main branch?
> > 
> > 
> > OK. In the main branch, which version in file `idf_component.yml` can I use, is `1.3.3` OK?
> 
> The next planned release version in main is 2.0.0, maybe you can use it in idf_component.yml?

OK.

**Pull request merged**
